src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -135,6 +156,49 @@ public static function fromDsn(string $dsn, array $redisOptions = [], \Redis $re
         return new self($configuration, $connectionCredentials, $redisOptions, $redis);
     }
 
+    private function claimOldPendingMessages()
+    {
+        try {
+            // This could soon be optimized with https://github.com/antirez/redis/issues/5212 or
+            // https://github.com/antirez/redis/issues/6256
+            $pendingMessages = $this->connection->xpending($this->stream, $this->group, '-', '+', 1);
+        } catch (\RedisException $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+
+        $claimableIds = [];
+        foreach ($pendingMessages as $pendingMessage) {
+            if ($pendingMessage[1] === $this->consumer) {
+                $this->couldHavePendingMessages = true;
+
+                return;
+            }
+
+            if ($pendingMessage[2] < (time() - $this->redeliverTimeout)) {
```

Comment:
The pendingMessage[2] does on my php-ext version (5.0) not return the timestamp as second param so I think we need to convert the id to timestamp. Can you check this on your used version?

```php
array(4) {
  [0]=>
  string(15) "1579730356351-0"
  [1]=>
  string(15) "other-consumer2"
  [2]=>
  int(1)
  [3]=>
  int(1)
}
```

Reply:
pendingMessage[2] is the idle time. I am sorry, I've missed that you added the `time()` in https://github.com/symfony/symfony/pull/35384#issuecomment-576964436 and just copied it in.

That's the time the message is pending, not the timestamp. Using a timestamp there (or using the redis id timestamp) would lead to enduser problems since the clock of the redis server would have to match the worker time.